crates/ruff/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -78,9 +90,44 @@ pub(crate) fn mutable_argument_default(checker: &mut Checker, arguments: &Argume
                 is_immutable_annotation(expr, checker.semantic())
             })
         {
-            checker
-                .diagnostics
-                .push(Diagnostic::new(MutableArgumentDefault, default.range()));
+            let mut diagnostic = Diagnostic::new(MutableArgumentDefault, default.range());
+
+            if checker.patch(diagnostic.kind.rule())
+                // Do not try to fix if the function is only one line.
+                && checker
+                    .locator
+                    .contains_line_break(TextRange::new(arguments.range().end(), body[0].start()))
+            {
+                // Set the default arg value to None
+                let arg_edit = Edit::range_replacement("None".to_string(), default.range());
+
+                // Add conditional check to set the default arg to its original value if still None
+                let mut check_lines = String::new();
+                let indentation = checker.locator.slice(TextRange::new(
+                    checker.locator.line_start(body[0].start()),
+                    body[0].start(),
+                ));
```

Comment:
You can use the `indentation_at_offset` helper here:

https://github.com/astral-sh/ruff/blob/3f1b489058d8be09b151563cc04b174c8eb2342b/crates/ruff_python_ast/src/whitespace.rs#L18